### PR TITLE
Switch review settings for operator, pkg, and common-files.

### PR DESCRIPTION
**Pull request by geeknoid**
Need to put the right constraints for tide-based repos.

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -98,14 +98,29 @@ branch-protection:
...
         pkg:
+          required_pull_request_reviews: &tide_reviews
+            require_code_owner_reviews: false
+          restrictions: &tide_restrictions
+            teams:
+              - repo-admins
           branches: *blocked_branches
         common-files:
+          required_pull_request_reviews: &tide_reviews
+            require_code_owner_reviews: false
```
Comment (howardjohn) [inline prow/config.yaml]: don't we want codeowners?

Comment (geeknoid) [reply prow/config.yaml]: Yes of course. Cut & paste syndrome.

What I really wanted with this PR is just restricting merges to admins.

File: prow/config.yaml [hunk 2]
```diff
@@ -98,14 +98,29 @@ branch-protection:
...
+              - repo-admins
           branches: *blocked_branches
         cni:
           branches: *blocked_branches
         installer:
           branches: *blocked_branches
         pkg:
+          required_pull_request_reviews: &tide_reviews
+            require_code_owner_reviews: true
+          restrictions: &tide_restrictions
```
Comment (fejta) [inline prow/config.yaml]: The tide_restrictions anchor should only be defined once. Thereafter we only need to point to it: `restrictions: *tide_restrictions` (the teams part is already included in the `*`).

If you need to extend the anchor then you need the
```yaml
restrictions:
  <<: *tide_restrictions
  my-extensions: whatever
```
pattern, but you don't here, so just `restrictions: *tide_restrictions` is right

File: prow/config.yaml [hunk 3]
```diff
@@ -98,14 +98,29 @@ branch-protection:
...
+            teams:
+              - repo-admins
           branches: *blocked_branches
         common-files:
+          required_pull_request_reviews: &tide_reviews
```
Comment (fejta) [inline prow/config.yaml]: This anchor has already been defined on L106/L112, so this should just be `required_pull_request_reviews: *tide_reviews`

File: prow/config.yaml [hunk 4]
```diff
@@ -98,14 +98,29 @@ branch-protection:
...
+            require_code_owner_reviews: true
+          restrictions: &tide_restrictions
+            teams:
+              - repo-admins
           branches: *blocked_branches
         istio:
           required_pull_request_reviews: &tide_reviews
```
Comment (fejta) [inline prow/config.yaml:120]: This should now become `required_pull_request_reviews: *tide_reviews`, dropping L112/L127

Same thing for L113/L128: should become `restrictions: *tide_restrictions` and not redefine the anchor

**Review by fejta — COMMENTED** [review]
LGTM intent, yaml anchor nits

**Comment by geeknoid** [comment]
PTAL

**Review by fejta — APPROVED** [review]

**Pull request merged**